code/datums/perks/oddity.dm

```diff
@@ -250,3 +250,31 @@
 				continue
 			H.adjustBruteLoss(-healing_power)
 			H.adjustFireLoss(-healing_power)
+
+/datum/perk/oddity/hive_born
+	name = "Hiveborn"
+	desc = "You feel electricty flow within your body to your hands. Powercells recharge in your hands."
+	icon_state = "circuitry"  //https://game-icons.net/1x1/lorc/circuitry.html
+	gain_text = "You feel a stabbing pain of something being injected into you, and with it a painfully pleaseant feeling of being improved."
+	var/cooldown = 10 SECONDS
+	var/initial_time
+	var/obj/item/weapon/cell/C
+
+/datum/perk/oddity/hive_born/assign(mob/living/carbon/human/H)
+	..()
+	initial_time = world.time
+
+/datum/perk/oddity/hive_born/on_process()
+	if(!..())
+		return
+	if(world.time < initial_time + cooldown)
+		return
+	initial_time = world.time
+	if((holder.l_hand && holder.l_hand != src && istype(holder.l_hand, /obj/item/weapon/cell)))
+		C = holder.l_hand
+		if(!C.fully_charged())
+			C.give(50)
+	if((holder.r_hand && holder.r_hand != src && istype(holder.r_hand, /obj/item/weapon/cell)))
+		C = holder.r_hand
+		if(!C.fully_charged())
+			C.give(50)
```

Comment:
Can this make batteries go over full charge?

Reply:
no, as the `if(!C.fully_charged())` checks if the powercell is fully charged

similary to how the recharger does

Reply:
Yeah, but if it's 97 charge out of 100, would it go to 147?

Reply:
> Yeah, but if it's 97 charge out of 100, would it go to 147?

i dont think so but, better safe than sorry, gonna quickly test

Reply:
![image](https://user-images.githubusercontent.com/59490776/118260534-be035580-b4b2-11eb-8dd7-18d69badfba1.png)
actually i think it won't because of how the `give` proc works

i can still test it out

Reply:
> Yeah, but if it's 97 charge out of 100, would it go to 147?

nope it doesn't, just tested